client/host_queue.go

```diff
@@ -169,46 +169,62 @@ func (q *queue) rotateOpsWithLock() []op {
 }
 
 func (q *queue) drain() {
-	wgAll := &sync.WaitGroup{}
+	var (
+		wgAll                       = &sync.WaitGroup{}
+		currWriteOpsByNamespace     = make(map[string][]op)
```

Comment:
Is there any new test for checking that writing two different namespaces will generate two fetch batch requests?  Thinking this might be a good idea to make sure going forward people never break this logic.